### DISCUSSION TITLE
River Runoff -

### DISCUSSION THREAD

**Discussion by armish2** [OpenDrift/opendrift#1067] [General]
Hello Team,

I am trying to learn particle modeling. I am trying to replicate the example 'river runoff'. 

"""
import os
from datetime import datetime, timedelta
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.dates import DateFormatter
import opendrift
from opendrift.models.oceandrift import OceanDrift
from opendrift.readers import reader_oscillating

outfile = 'runoff.nc'  # Raw simulation output
histogram_file = 'runoff_histogram.nc'

o = OceanDrift(loglevel=20)
"""
Error : 
12:08:12 INFO    opendrift.models.basemodel:241: Could not import iomodule netcdf
Traceback (most recent call last):

  Cell In[4], line 17
    o = OceanDrift(loglevel=20)

  File C:\opendrift\opendrift\models\oceandrift.py:115 in __init__
    super(OceanDrift, self).__init__(*args, **kwargs)

  File C:\opendrift\opendrift\models\basemodel.py:242 in __init__
    self.io_init = types.MethodType(io_module.init, self)

UnboundLocalError: cannot access local variable 'io_module' where it is not associated with a value.

In basemodel.py file, I tried to assign a value for 'io_module ' variable. Still I am facing error.

**Comment by knutfrode** [comment]
It seems there is something wrong with your installation. Did you follow the steps on https://opendrift.github.io/install.html ?

It is probably best to not make any modifications in basemodel.py, or any of the core files.
You can check with `git status` what has been modified locally.

To test your installation, you can run `pytest` from the main opendrift folder.

**Comment by armish2** [comment]
Thank you for getting back to me. 
Yes, I followed the same steps  - https://opendrift.github.io/install.html 

when I run pytest from miniconda prompt, I get error - 
"""
FAILED tests/models/test_run.py::TestRun::test_seed_on_land - PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'out.nc'
FAILED tests/models/test_run.py::TestRun::test_seed_shapefile - TypeError: Wrong number or type of arguments for overloaded function 'Geometry_Transform'.
FAILED tests/models/openoil/test_oillibrary.py::test_seed_metadata_output - PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'openoil.nc'
FAILED tests/readers/test_interpolation.py::TestInterpolation::test_dateline - PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'opendrift_test_winds_180_180.nc'
FAILED tests/readers/test_xarray.py::TestXarray::test_density - PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'test_xarray.nc'
===================================================================================================================== 5 failed, 189 passed, 19 skipped, 2 xfailed, 1941 warnings in 290.35s (0:04:50) ===

"""

I am using python version - 3.11.0
